### PR TITLE
Fail when defining a message attribute multiple times

### PR DESCRIPTION
**Pull request by bforma**

### CODE REVIEW

**Review by lvonk — COMMENTED** [review]
Good to validate this. Can we somehow make it so it is done only at load time (or the frst time `types` is called?)

File: lib/sequent/core/helpers/attribute_support.rb [hunk 1]
```diff
@@ -42,17 +42,19 @@ class UnknownAttributeError < StandardError; end
         module ClassMethods
           def types
             @types ||= {}
-            return @merged_types if @merged_types
 
-            @merged_types = is_a?(Class) && superclass.respond_to?(:types) ? @types.merge(superclass.types) : @types
+            merged_types = is_a?(Class) && superclass.respond_to?(:types) ? @types.merge(superclass.types) : @types
```
[the remark this answers is not in the document]
Comment (bforma) [reply lib/sequent/core/helpers/attribute_support.rb:46]: Sounds dangerous, it already had a bug...

**Comment by bforma** [comment]
> Good to validate this. Can we somehow make it so it is done only at load time (or the frst time `types` is called?)

I don't understand what you mean with 'make it so it is done only at load time'.

Validation is only done (once) when you define attributes with `attrs`.

**Comment by bforma** [comment]
> Since the early return is now removed it will traverse on each deserialization?

I don't know enough about the internal workings to conclude that. This at least fixes a bug I encountered. The 'attribute support' code is a bit too complicated for me to follow. Perhaps we can have a look together later this week?

**Comment by lvonk** [comment]
> > Since the early return is now removed it will traverse on each deserialization?
> 
> I don't know enough about the internal workings to conclude that. This at least fixes a bug I encountered. The 'attribute support' code is a bit too complicated for me to follow. Perhaps we can have a look together later this week?

Yes, this old code definitely needs some ❤️ 😄

**Review by lvonk — APPROVED** [review]

**Pull request merged**